Getting amqp (Hex package) ** (ArgumentError) argument error

While running **make all** step for RabbitMQ Server 3.7.0, I am getting the following error:

`Putting child 0x95a5d000 (escript/rabbitmqctl) PID 20257 on the chain.
Live child 0x95a5d000 (escript/rabbitmqctl) PID 20257
 GEN    escript/rabbitmqctl
Resolving Hex dependencies...
Dependency resolution completed:
  amqp 0.2.4
  csv 2.0.0
  dialyxir 0.5.1
  json 1.0.2
  parallel_stream 1.0.5
  simetric 0.2.0
  temp 0.4.3
* Getting amqp (Hex package)
** (ArgumentError) argument error
    (stdlib) re.erl:734: :re.run("https://repo.hex.pm/tarballs/amqp-0.2.4.tar", {:re_pattern, 1, 0, 0, <<69, 82, 67, 80, 91, 0, 0, 0, 0, 0, 0, 0, 81, 0, 0, 0, 255, 255, 255, 255, 255, 255, 255, 255, 47, 0, 64, 0, 0, 0, 1, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...>>}, [{:capture, :all, :index}, :global])
    (elixir) lib/regex.ex:590: Regex.do_replace/4
    (hex) lib/hex/scm.ex:103: Hex.SCM.update/1
    (hex) lib/hex/scm.ex:150: Hex.SCM.checkout/1
    (mix) lib/mix/dep/fetcher.ex:64: Mix.Dep.Fetcher.do_fetch/3
    (mix) lib/mix/dep/converger.ex:180: Mix.Dep.Converger.all/9
    (mix) lib/mix/dep/converger.ex:189: Mix.Dep.Converger.all/9
    (mix) lib/mix/dep/converger.ex:119: Mix.Dep.Converger.all/7
Reaping losing child 0x95a5d000 PID 20257
Makefile:73: recipe for target 'escript/rabbitmqctl' failed
make: *** [escript/rabbitmqctl] Error 1
Removing child 0x95a5d000 PID 20257 from chain.`

Facing similar issue on rabbitmq 3.7.3 with hex.info as 
Hex:    0.17.3
Elixir: 1.6.1
OTP:    20.0.4

Built with: Elixir 1.6.1 and OTP 20.0.4

output:
Dependency resolution completed:
  amqp 0.2.4
  csv 2.0.0
  dialyxir 0.5.1
  json 1.0.2
  parallel_stream 1.0.6
  simetric 0.2.0
  temp 0.4.4
* Getting json (Hex package)
** (ArgumentError) argument error
    (stdlib) re.erl:734: :re.run("https://repo.hex.pm/tarballs/json-1.0.2.tar", {:re_pattern, 1, 0, 0, <<69, 82, 67, 80, 91, 0, 0, 0, 0, 0, 0, 0, 81, 0, 0, 0, 255, 255, 255, 255, 255, 255, 255, 255, 47, 0, 64, 0, 0, 0, 1, 0, 0, 0, 64, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, ...>>}, [{:capture, :all, :index}, :global])
    (elixir) lib/regex.ex:598: Regex.do_replace/4
    (hex) lib/hex/scm.ex:103: Hex.SCM.update/1
    (hex) lib/hex/scm.ex:163: Hex.SCM.checkout/1
    (mix) lib/mix/dep/fetcher.ex:64: Mix.Dep.Fetcher.do_fetch/3
    (mix) lib/mix/dep/converger.ex:185: Mix.Dep.Converger.all/9
    (mix) lib/mix/dep/converger.ex:121: Mix.Dep.Converger.all/7
    (mix) lib/mix/dep/converger.ex:106: Mix.Dep.Converger.all/4
Makefile:85: recipe for target 'escript/rabbitmqctl' failed

--- a/lib/hex/scm.ex
+++ b/lib/hex/scm.ex
@@ -106,7 +106,6 @@ defmodule Hex.SCM do
     filename = "#{name}-#{lock.version}.tar"
     path = cache_path(repo, filename)
     url = Hex.Repo.get_repo(repo).url <> "/tarballs/#{filename}"
-    safe_url = Regex.replace(~r/\/\/([^:]*):[^@]+@/, url, "//\\1:******@")
 
     case Hex.Parallel.await(:hex_fetcher, {:tarball, repo, name, lock.version}, @fetch_timeout) do
       {:ok, :cached} ->
@@ -122,12 +121,14 @@ defmodule Hex.SCM do
           Registry.persist()
         end
 
+        safe_url = prune_uri_userinfo(url)
         Hex.Shell.debug("  Fetched package (#{safe_url})")
 
       {:error, reason} ->
         Hex.Shell.error(reason)
 
         unless File.exists?(path) do
+          safe_url = prune_uri_userinfo(url)
           Mix.raise("Package fetch failed and no cached copy available (#{safe_url})")
         end
 
@@ -158,6 +159,13 @@ defmodule Hex.SCM do
       |> Enum.sort()
 
     {:hex, String.to_atom(lock.name), lock.version, lock.checksum, managers, deps, lock.repo}
+  end
+
+  defp prune_uri_userinfo(url) do
+    case URI.parse(url) do
+      %URI{userinfo: nil} -> url
+      uri -> URI.to_string(%{uri | userinfo: "******"})
+    end
   end
 
   def checkout(opts) do

